src/main/java/org/sonatype/repository/conan/internal/proxy/matcher/ConanMatcherDeserializer.java

```diff
@@ -0,0 +1,31 @@
+package org.sonatype.repository.conan.internal.proxy.matcher;
+
+import java.io.IOException;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+
+/**
+ * Based on the configuration will create a {@link BintrayMatcher} or {@link StandardMatcher}
+ *
+ * @since 0.0.2
+ */
+public class ConanMatcherDeserializer
```

Comment:
As mentioned, this seems like a lot for what could just be a factory method that takes an enum. It's not bad, and obviously in some cases you need to do stuff like this, but to me it feels like added complexity that someone else may or may not understand. Particularly for something like a formats implementation where we (hopefully) get developers who know enough Java to be dangerous but perhaps spend most of their time in C++. The more obvious something is the easier it is for someone else to jump on board and contribute.

Reply:
Separate ticket raised to consider this approach https://github.com/sonatype-nexus-community/nexus-repository-conan/issues/17